lab-3.py

```diff
@@ -6,25 +6,45 @@
 s3 = boto3.client('s3')
 
 def main(argv):
-    cmd = argv
+
+    cmd = determine_command(int(sys.argv[1]))
     log_file_name = datetime.datetime.now(datetime.timezone.utc).strftime("%m_%d_%Y") + "_logfile"
     kickoff_subprocess(cmd, log_file_name)
     upload_output_to_S3(log_file_name)
 
+# instead of allowing free-form input
+# we instead have them enter a number
+# which corresponds to a command that is pre-written
+# you also could present a menu that allows them to select the command
+# but either way we are avoiding allowing free-form entry
+def determine_command(user_input):
+    if(user_input == 1):
+        return ['echo', "command1"]
+    if(user_input == 2):
+        return ['echo', "command2"]
+    if(user_input == 3):
+        return ['echo', "command3"]
+    else:
+        return ['echo', "default"]
+
 def kickoff_subprocess(cmd, log_file_name):
+    #Linux
     process = subprocess.call(cmd, shell=False)
-    # not needed as with open() used
-    # file = open(log_file_name, "a+")
+
+    #Windows
+    #process = subprocess.call(['cmd', '/c'], shell=False)
+
+    file = open(log_file_name, "a+")

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *file*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *file* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).